cl_manycore/libraries/bsg_manycore_tile.cpp

```diff
@@ -55,51 +60,259 @@ int hb_mc_tile_unfreeze (uint8_t fd, uint8_t x, uint8_t y) {
+		return HB_MC_FAIL;
+	}
+
+	eva_t bsg_tile_group_id_x_eva, bsg_tile_group_id_y_eva;
+	if (symbol_to_eva(elf, "__bsg_tile_group_id_x", &bsg_tile_group_id_x_eva) != HB_MC_SUCCESS){
+		fprintf(stderr, "hb_mc_tile_set_tile_group_id_symbols() --> symbol_to_eva(): failed to aquire __bsg_tile_group_id_x eva.\n");
+		return HB_MC_FAIL;
+	}
+	if (symbol_to_eva(elf, "__bsg_tile_group_id_y", &bsg_tile_group_id_y_eva) != HB_MC_SUCCESS){
+		fprintf(stderr, "hb_mc_tile_set_tile_group_id_symbols() --> symbol_to_eva(): failed to aquire __bsg_tile_group_id_y eva.\n");
+		return HB_MC_FAIL;
+	}
+
+	if (hb_mc_copy_to_epa(fd, x, y, bsg_tile_group_id_x_eva >> 2 /* TODO: magic number */, &tg_id_x, 1) != HB_MC_SUCCESS) {
+		fprintf(stderr, "hb_mc_tile_set_tile_group_id_symbols() --> hb_mc_copy_to_epa(): failed to set tile __bsg_tile_group_id_x symbol.\n"); 
+		return HB_MC_FAIL;
+	}
+	#ifdef DEBUG
+		fprintf(stderr, "Setting tile (%d,%d) __bsg_tile_group_id_x (eva 0x%x) to %d.\n", x, y, bsg_tile_group_id_x_eva, tg_id_x);
+	#endif
 
-	if (rc != HB_MC_SUCCESS) {
+	if (hb_mc_copy_to_epa(fd, x, y, bsg_tile_group_id_y_eva >> 2 /* TODO: magic number */, &tg_id_y, 1) != HB_MC_SUCCESS) {
+		fprintf(stderr, "hb_mc_tile_set_tile_group_id_symbols() --> hb_mc_copy_to_epa(): failed to set tile __bsg_tile_group_id_y symbol.\n"); 
 		return HB_MC_FAIL;
 	}
+	#ifdef DEBUG
+		fprintf(stderr, "Setting tile (%d,%d) __bsg_tile_group_id_y (eva 0x%x) to %d.\n", x, y, bsg_tile_group_id_y_eva, tg_id_y);
+	#endif
 
 	return HB_MC_SUCCESS;
 }
+
+/*! 
+ * Sets a Vanilla Core Endpoint's tile's __bsg_grid_dim_x/y symbol.
+ * @param[in] fd userspace file descriptor
+ * @param[in] eva_id eva-to-npa mapping
+ * @param[in] x x coordinate of tile
+ * @param[in] y y coordinate of tile
+ * @param[in] grid_dim_x/y tile group's grid dimensions.
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure.
+* */
+int hb_mc_tile_set_grid_dim_symbols (uint8_t fd, eva_id_t eva_id, char* elf,  uint32_t x, uint32_t y, uint32_t grid_dim_x, uint32_t grid_dim_y){
+	if (hb_mc_fifo_check(fd) != HB_MC_SUCCESS) {
+		fprintf(stderr, "hb_mc_tile_set_grid_dim_symbols(): invalid device %d.\n", fd);
+		return HB_MC_FAIL;
+	}
+
+	eva_t bsg_grid_dim_x_eva, bsg_grid_dim_y_eva;
+	if (symbol_to_eva(elf, "__bsg_grid_dim_x", &bsg_grid_dim_x_eva) != HB_MC_SUCCESS){
+		fprintf(stderr, "hb_mc_tile_set_grid_dim_symbols() --> symbol_to_eva(): failed to aquire __bsg_grid_dim_x eva.\n");
+		return HB_MC_FAIL;
+	}
+
+	if (symbol_to_eva(elf, "__bsg_grid_dim_y", &bsg_grid_dim_y_eva) != HB_MC_SUCCESS){
+		fprintf(stderr, "hb_mc_tile_set_grid_dim_symbols() --> symbol_to_eva(): failed to aquire __bsg_grid_dim_y eva.\n");
+		return HB_MC_FAIL;
+	}
+
+	if (hb_mc_copy_to_epa(fd, x, y, bsg_grid_dim_x_eva >> 2 /* TODO: magic number */, &grid_dim_x, 1) != HB_MC_SUCCESS) {
+		fprintf(stderr, "hb_mc_tile_set_grid_dim_symbols() --> hb_mc_copy_to_epa(): failed to set tile __bsg_grid_dim_x symbol.\n"); 
```

Comment:
* I recommend using the automatic variable `__func__`  instead of hard coding the string.

Reply:
Good point! will do, thanks!